whois-query/src/main/java/net/ripe/db/whois/query/planner/AbuseCFinder.java

```diff
@@ -28,23 +31,49 @@
     private final Ipv4Tree ipv4Tree;
     private final Ipv6Tree ipv6Tree;
     private final Maintainers maintainers;
+    private final AbuseValidationStatusDao abuseValidationStatusDao;
 
     @Autowired
     public AbuseCFinder(final RpslObjectDao objectDao,
                         final Ipv4Tree ipv4Tree,
                         final Ipv6Tree ipv6Tree,
-                        final Maintainers maintainers) {
+                        final Maintainers maintainers,
+                        final AbuseValidationStatusDao abuseValidationStatusDao) {
         this.objectDao = objectDao;
         this.ipv4Tree = ipv4Tree;
         this.ipv6Tree = ipv6Tree;
         this.maintainers = maintainers;
+        this.abuseValidationStatusDao = abuseValidationStatusDao;
     }
 
-    @CheckForNull
-    @Nullable
-    public String getAbuseContact(final RpslObject object) {
+    public Optional<AbuseContact> getAbuseContact(final RpslObject object) {
         final RpslObject role = getAbuseContactRole(object);
-        return (role != null) ? role.getValueForAttribute(AttributeType.ABUSE_MAILBOX).toString() : null;
+        return role != null?
+                Optional.of(new AbuseContact(
+                        role.getKey(),
+                        role.getValueForAttribute(AttributeType.ABUSE_MAILBOX),
+                        abuseValidationStatusDao.isSuspect(role.getValueForAttribute(AttributeType.ABUSE_MAILBOX)),
+                        getOrgToContact(object)
+                )) : Optional.empty();
+    }
+
+    @Nullable
+    private CIString getOrgToContact(final RpslObject object) {
+        if (object.containsAttribute(AttributeType.SPONSORING_ORG)) {
```

Comment:
Is the sponsoring org only needed for suspect abuse-c addresses? Otherwise it's the closest org-id?

Reply:
ok